jax/_src/cudnn/fused_attention_stablehlo.py

```diff
@@ -604,22 +678,59 @@ def _infer_bwd_output_sharding(mesh, arg_shapes, variadic_args):
   out_shardings = (grad_query_sharding, grad_key_sharding, grad_value_sharding)
   return out_shardings
 
-_dot_product_attention_bwd_lower = custom_partitioning(_dot_product_attention_bwd_impl, static_argnums=(8,9,10,11,12,13))
-def _dot_product_attention_bwd_infer_sharding_from_operands(scale, seed, dropout_rate, variadic_args, is_flash_attention, is_causal_mask, mesh, arg_shapes, result_shape):
+_dot_product_attention_bwd_lower = custom_partitioning(
+    _dot_product_attention_bwd_impl, static_argnums=(8, 9, 10, 11, 12, 13, 14)
+)
+
+def _dot_product_attention_bwd_infer_sharding_from_operands(
+    scale, seed, dropout_rate, variadic_args, is_flash_attention,
+    is_causal_mask, layout, mesh, arg_shapes, result_shape):
   return _infer_bwd_output_sharding(mesh, arg_shapes, variadic_args)
 
-def _dot_product_attention_bwd_partition(scale, seed, dropout_rate, variadic_args, is_flash_attention, is_causal_mask, mesh, arg_shapes, result_shape):
+def _dot_product_attention_bwd_partition(
+    scale, seed, dropout_rate, variadic_args, is_flash_attention,
+    is_causal_mask, layout, mesh, arg_shapes, result_shape):
   out_shardings = _infer_bwd_output_sharding(mesh, arg_shapes, variadic_args)
   # args sharding
   arg_shardings = tuple([arg_i.sharding for arg_i in arg_shapes])
-  impl = partial(_dot_product_attention_bwd_impl, scale=scale, seed=seed, dropout_rate=dropout_rate,
-                variadic_args=variadic_args, is_flash_attention=is_flash_attention, is_causal_mask=is_causal_mask)
+  impl = partial(
+      _dot_product_attention_bwd_impl, scale=scale, seed=seed,
+      dropout_rate=dropout_rate, variadic_args=variadic_args,
+      is_flash_attention=is_flash_attention, is_causal_mask=is_causal_mask,
+      layout=layout,
+  )
   return mesh, impl, out_shardings, arg_shardings
 
+# We redefine the xla.apply_primitive here since we need to set the jit option
+# static_argnames='layout' to ensure that the layout (str) is static.
+def apply_primitive(prim, *args, **params):
```

Comment:
why is this required?

Reply:
It is mainly to specify the `static_argnames` in api.jit(). Otherwise, the compiler will complain the str is not a supported jax type.